### PR TITLE
FIP: gas model adjustment for user programmability.

### PR DESCRIPTION
**Pull request by raulk**
This FIP follows #315, and introduces further gas model changes to address various aspects of user programmability.

### CODE REVIEW

File: FIPS/fip-nnnn.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
+---
+fip: "nnnn"
+title: Gas model adjustment for for user programmability
+author: Raúl Kripalani (@raulkr), Steven Allen (@stebalien)
```
Comment (anorth) [inline FIPS/fip-nnnn.md:4]: Typo in `@raulkr` (also in fip-0032).

**Review by anorth — COMMENTED** [review]
It looks like this PR accidentally duplicates part of #315. I've ignored the document `fip-0032.md` in this review.

File: FIPS/fip-nnnn.md (hunk 1 above)
Comment (raulk) [reply FIPS/fip-nnnn.md:4]: Thanks for noticing this. We really need a unified identity system :D

File: FIPS/fip-nnnn.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
+---
+fip: "nnnn"
```
Comment (kaitlin-beegle) [inline FIPS/fip-nnnn.md:2]: FIP0032, of course!

**Comment by kaitlin-beegle** [comment]
Now that we're initiating Last Call on FVM FIPs 30-32, let's merge this FIP in as well in anticipation of upcoming work. 

FIP Editors, would you mind adding your review of the initial draft? @jennijuju @arajasek @momack2 

@raulk, it looks as if this branch should easily merge into master now that #315 is merged, but lmk if you have concerns about the way the PR was set up.

File: FIPS/fip-nnnn.md (hunk 2 above)
Comment (kaitlin-beegle) [inline FIPS/fip-nnnn.md:2]: FIP0037!

**Review by kaitlin-beegle — CHANGES_REQUESTED** [review]

**Pull request merged**